ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/ReplLoadTask.java

```diff
@@ -166,14 +185,24 @@ a database ( directory )
           }
           loadTaskTracker.update(functionsTracker);
           functionsTracker.debugLog("functions");
+
+          // Add ReplStateLogTask as function load tasks are always fully added in current cycle
+          loadTaskTracker.updateTaskCount(createFunctionReplLogTask(functionsTracker.tasks(),
+                  iterator.replLogger(), loadFunction.functionName()));
```

Comment:
all the tasks that are being added on table / table / function completion should happening in side the task creators for those objects, since all the info is available there.